src/containers/EditSubjectFrontpage/components/SubjectpageForm.tsx

```diff
@@ -0,0 +1,193 @@
+/**
+ * Copyright (c) 2020-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree. *
+ */
+
+import React, { FC, useState } from 'react';
+import { injectT } from '@ndla/i18n';
+import { Formik, Form, FormikProps } from 'formik';
+import { Footer } from '@ndla/editor';
+import { SubjectpageEditType, TranslateType } from '../../../interfaces';
+import SimpleLanguageHeader from '../../../components/HeaderWithLanguage/SimpleLanguageHeader';
+import { FormikAlertModalWrapper, formClasses } from '../../FormikForm';
+import validateFormik from '../../../components/formikValidationSchema';
+import { isFormikFormDirty, subjectpageRules } from '../../../util/formHelper';
+import { toEditSubjectpage } from '../../../util/routeHelpers';
+import usePreventWindowUnload from '../../FormikForm/preventWindowUnloadHook';
+import SubjectpageAccordionPanels from './SubjectpageAccordionPanels';
+import { useSubjectpageFormHooks } from '../../FormikForm/subjectpageFormHooks';
+import {
+  editorValueToPlainText,
+  plainTextToEditorValue,
+} from '../../../util/articleContentConverter';
+import SaveButton from '../../../components/SaveButton';
+
+interface Props {
+  t: TranslateType;
+  subjectpage: SubjectpageEditType;
+  updateSubjectpage: Function;
+  selectedLanguage: string;
+  subjectId: number;
+  isNewlyCreated: boolean;
+}
+
+const getInitialValues = (
+  subjectpage: SubjectpageEditType,
+  subjectId: number,
+  selectedLanguage: string,
+) => {
+  return {
+    articleType: 'subjectpage',
+    supportedLanguages: subjectpage.supportedLanguages || [],
+    language: selectedLanguage,
+    description: plainTextToEditorValue(subjectpage.description, true),
+    title: subjectpage.title || '',
+    mobileBanner: subjectpage.mobileBanner || undefined,
+    desktopBanner: subjectpage.desktopBanner || undefined,
+    visualElement: subjectpage.visualElement || {
+      resource: '',
+      url: '',
+      resource_id: '',
+    },
+    visualElementAlt: subjectpage.visualElementAlt || '',
+    editorsChoices: subjectpage.editorsChoices || [],
+    facebook: subjectpage.facebook || '',
+    filters: subjectpage.filters || [],
+    goTo: subjectpage.goTo || [],
+    id: subjectpage.id,
+    latestContent: subjectpage.latestContent || [],
+    layout: subjectpage.layout || '',
```

Comment:
Dette er en enum så vi må sette dette til en standardverdi.
```suggestion
    layout: subjectpage.layout || 'single',
```